### PR TITLE
Initial prometheus chart supporting data capture and service discovery.

### PR DESCRIPTION
**Pull request by davidewatson**
Note that this checkin also removes promdash now that we use grafana.

### CODE REVIEW

**Comment by davidewatson** [comment]
Please do not merge this yet as I am still making changes. In particular, PVCs, and a generic scrape config.

File: prometheus/values.yaml [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 color: E5CCFF
-dburl: "sqlite3:/promdash/file.sqlite3"
-promdash_image: "quay.io/samsung_cnct/promdash:latest"
-promdash_port: 3000
-promdash_node_port: 30361
-prometheus_image: "quay.io/samsung_cnct/promdash:latest"
+prometheus_image: "prom/prometheus:v1.2.1"
 prometheus_port: 9090
 prometheus_node_port: 30360
+storage_local_retention: '876000h'
```
Comment (spiffxp) [inline prometheus/values.yaml]: this looks to be unused?

File: prometheus/values.yaml [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-pushgateway_image: "prom/pushgateway:latest"
+kubelet_cadvisor_target: "127.0.0.1:11111"
+pushgateway_image: "prom/pushgateway:0.3.0"
 pushgateway_port: 9091
 pushgateway_node_port: 30362
-sidecar_image: "lachlanevenson/k8s-kubectl"
+config_dir: /etc/prometheus/
+data_dir: /prometheus
+storage_class: anything
+storage_size: 1Gi
```
Comment (spiffxp) [inline prometheus/values.yaml]: I'm guessing this means we allocate a 1Gi volume to prometheus to store its data, any ballpark on what that works out to in terms of volume of data / retention?  or is this just an arbitrary default (which is also fine, I just think the magic number needs a comment)

File: prometheus/Chart.yaml [hunk 3]
```diff
@@ -1,3 +1,4 @@
+---
 name: prometheus
 description: Prometheus Helm chart for Kubernetes
 version: 0.1.0
```
Comment (spiffxp) [inline prometheus/Chart.yaml]: if you're changing the chart, change the version

File: prometheus/values.yaml [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
-dburl: "sqlite3:/promdash/file.sqlite3"
-promdash_image: "quay.io/samsung_cnct/promdash:latest"
-promdash_port: 3000
-promdash_node_port: 30361
-prometheus_image: "quay.io/samsung_cnct/promdash:latest"
+prometheus_image: "prom/prometheus:v1.2.1"
 prometheus_port: 9090
 prometheus_node_port: 30360
+storage_local_retention: '876000h'
 kube_apiserver: "localhost:8001"
```
Comment (davidewatson) [inline prometheus/values.yaml]: Done.

**Comment by spiffxp** [comment]
I'm not sure if @coffeepac's comment was addressed or not, I'll leave it to him to lgtm.  Otherwise I'm happy.

**Comment by davidewatson** [comment]
It was the same comment you had about removing unused template variables...

**Comment by spiffxp** [comment]
@davidewatson right, I don't think you've removed the things he was referring to (eg: `kube_apiserver`)

**Comment by coffeepac** [comment]
LGTM

**Pull request merged**